graphx/src/main/scala/org/apache/spark/graphx/lib/StronglyConnectedComponents.scala

```diff
@@ -44,6 +44,11 @@ object StronglyConnectedComponents {
     // graph we are going to work with in our iterations
     var sccWorkGraph = graph.mapVertices { case (vid, _) => (vid, false) }.cache()
 
+    // helper variables to unpersist cached graphs
+    var prevSccGraph1 = sccGraph
+    var prevSccGraph2 = sccGraph
```

Comment:
I guess I could avoid to have this two vars, but it was hard to reason about it code with one only. One sccGraph is persisted inside do-while loop, and the second at the end of outside while loop. With two variables it is clearer which graph is unpersisted. For example when do-while loop is left you have to unpersist prevSccGraph1, but not necessarily prevSccGraph2.